charms/istio-pilot/src/charm.py

```diff
@@ -112,6 +119,19 @@ def __init__(self, *args):
         # available, revoked, invalidated, or if the certs relation is broken
         self.framework.observe(self._cert_handler.on.cert_changed, self.reconcile)
 
+        # ---- Start of the block
+        # ---- WARNING: this feature is not recommended, but is supported in 1.17 and 1.18.
+        # ---- For details please refer to canonical/istio-operators#380.
+        # ---- FIXME: Remove this block after releasing 1.18.
+        # Save SSL information and reconcile
+        self.framework.observe(self.on.set_tls_manually_action, self.save_tls_secret)
+        self.framework.observe(self.on.secret_changed, self.reconcile_tls_secret)
```

Comment:
Can we use `self.reconcile` here for handling changes in tls secrets instead of adding a `reconcile_tls_secret`?  The rest of this charm does a full reconcile on any meaningful event to avoid us forgetting about some specific interaction between events and I could see it being helpful here as well.  There's also some failure logic in the reconcile function that we'll lose if we avoid it.

This also applies to [this line](https://github.com/canonical/istio-operators/pull/394/files#diff-d3a4e684bd1a3a1822df5ab7605cfce00c97cd7b52f44ed770526c4f4f80eacdR132)